### DISCUSSION TITLE
Isthmus upgrade: Hardfork Planning

### DISCUSSION THREAD

**Discussion by BlocksOnAChain** [ethereum-optimism/specs#373] [General]
- We will use this discussion to do a public "brainstorm" about the potential scope and ideas that we want to include for the isthmus hardfork that follows Holocene. This fork is targeting Governance Cycle 33, which starts of February 6th.
- Feel free to share your ideas and specifications with the community, so we have a great public planning session for our upcoming network upgrade

**Comment by yuwen01** [comment]
I propose adding another transaction fee, called the `OperatorFee`. This fee allows new OP chain variants to account for their unique cost structure. For example, the existing fee structure isn't friendly to chains using alt-DA's, since the `l1fee` only accounts for Ethereum's `blobGasFee` instead of an alt-DA's fee. 

[Design Doc](https://github.com/ethereum-optimism/design-docs/pull/81)
[Spec](https://github.com/ethereum-optimism/specs/pull/382)

[REVM implementation WIP](https://github.com/bluealloy/revm/pull/1798)
[OP-Geth implementation WIP](https://github.com/ethereum-optimism/op-geth/pull/388/)
[Optimism monorepo implementation WIP](https://github.com/ethereum-optimism/optimism/pull/12166)
[OP-alloy implementation WIP](https://github.com/alloy-rs/op-alloy/pull/130)

**Reply by tynes** [reply]
Definitely in favor of including this

**Reply by BlocksOnAChain** [reply]
I support this suggestion and I think we should include it for Isthmus hardfork.

**Comment by tynes** [comment]
I propose that we include the "Standard L2 Genesis" project, the [design doc](https://github.com/ethereum-optimism/design-docs/pull/97) can be found here

**Comment by mds1** [comment]
I propose that we include an upgrade of all L1 and L2 Solidity contracts to 0.8.25, as agreed upon in https://github.com/ethereum-optimism/design-docs/pull/106

**Comment by tynes** [comment]
I propose that we include the Pectra changes into the OP Stack as part of Isthmus. Getting EIP-7702 early is important for bleeding edge applications to deploy to OP Stack chains.

See https://eips.ethereum.org/EIPS/eip-7600 for scope

- Span batch changes are required for 7702, see https://github.com/ethereum-optimism/specs/issues/318
- We should make sure that the BLS precompiles aren't too expensive in the fault proof

From @ajsutton 

> For whenever we pull Pectra into the OP stack, we will need to do some benchmarking on the new BLS precompiles - there's a pretty high likelihood we'll need to apply the preimage oracle acceleration to them.  Since the G1/G2 MSM precompile pairing check operation has a variable gas cost if it needs to go via the preimage oracle we will need to include a maximum input length for the precompile to ensure any call to it on L2 can be executed on L1 with some gas to spare to store the result in the preimage oracle.

**Comment by vdamle** [comment]
I propose that we include the `L2 withdrawals root in block header` feature in `Isthmus` (it was de-scoped from `Holocene`)

ref: https://github.com/ethereum-optimism/optimism/issues/12044

WIP:

https://github.com/ethereum-optimism/specs/pull/396
https://github.com/ethereum-optimism/op-geth/pull/383
https://github.com/ethereum-optimism/op-geth/pull/399

**Comment by mds1** [comment]
I propose that we include the Custom Gas Token (CGT) code.

The standard config does not change, and still requires ETH as the gas token to be a standard chain. Therefore, functionally, there is no change by including this feature. The rationale is that CGT code is currently coupled with other code, and it's less work to just ship the CGT as a regular feature (and have it part of OPCM, etc) and require that standard chains don't use this feature, than to decouple the code

**Comment by maurelian** [comment]
I propose that we add changes to the L1 contracts required to enable [L1 upgrades via the OPCM](https://github.com/ethereum-optimism/design-docs/pull/159).

**Comment by benjaminion** [comment]
Closing this discussion as it's had a good run and we need to lock down scope for Isthmus at some point.
